Feature/addition rest endpoints

lliss opened the pull request:
Connects #7 

Builds off of https://github.com/ProboCI/probo-asset-receiver/pull/6 https://github.com/ProboCI/probo-asset-receiver/pull/8 and https://github.com/ProboCI/probo-asset-receiver/pull/9

Adds additional API endpoints needed to allow users to review the size of their assets and remove them if they so desire.

NOTE: Only the last 5 commits are part of this PR.

### To Test:
 - You will first need to run the migrations (see testing instructions https://github.com/ProboCI/probo-asset-receiver/pull/6)
 - Upload new files
 - View assets in your bucket at `/buckets/:bucketName/assets`
 - Ensure the sizes and other attributes are appropriate for the file you uploaded.
 - Delete the file via the new endpoint `/buckets/:bucketName/assets/:assetName`
 - Ensure that the db keys and the files on disk are deleted.

tortillaj commented on lib/Server.js:
```diff
@@ -1,28 +1,30 @@
 var Server = function(options) {
   this.options = options || {
     encryptionCipher: 'aes-256-cbc',
     encryptionPassword: 'SECRET',
```
Let's make this `encryptionPassword` a config option.

tortillaj replied:
Oh I see now, never mind this comment ^

tortillaj commented on lib/Server.js:
```diff
@@ -77,6 +75,9 @@ Server.prototype.setup = function() {
   app.delete('/buckets/:bucket/token/:token', auth, this.routes.deleteBucketToken);
   app.get('/buckets/:bucket/token', auth, this.routes.listBucketTokens);
   app.get('/asset/:bucket/:assetName', auth, this.routes.serveFileAsset);
+  app.get('/asset-size/:bucket/:assetName', auth, this.routes.listAssetSize);
```
This is kind of nit-picky, but I'd like to propose `/buckets/:bucket/asset/:assetName/size` just to make the URL more predictable.

lliss replied:
I'm fine with that.

tortillaj commented on test/awsS3-test.js:
```diff
@@ -0,0 +1,77 @@
+'use strict';
```
this file needs linting

lliss replied:
Ok.

tortillaj commented on migrations/probo-asset-fix-and-upload:
```diff
@@ -0,0 +1,104 @@
+#! /usr/bin/env node
```
Can we rename the migrations to include a number indicating the order they need to be run? e.g. `0001-probo-asset-fix-and-upload`

tizzo replied:
Yeah, we should do that. Also - I think we should probably title this
migration to be a bit more descriptive. I think this is
"0000-get-upload-sizes-and-reencrypt' or something?

On Thu, Mar 3, 2016 at 8:57 AM, James Cole <notifications@github.com> wrote:

> In migrations/probo-asset-fix-and-upload
> <https://github.com/ProboCI/probo-asset-receiver/pull/10#discussion_r54882103>
> :
>
> > @@ -0,0 +1,104 @@
> > +#! /usr/bin/env node
>
> Can we rename the migrations to include a number indicating the order they
> need to be run? e.g. 0001-probo-asset-fix-and-upload
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ProboCI/probo-asset-receiver/pull/10/files#r54882103>.
>

lliss replied:
Sure. No problem.

tortillaj commented on lib/Server.js:
```diff
@@ -69,6 +62,11 @@ Server.prototype.setup = function() {
   });
 
+
+  app.get('/', function(req, res) {
```
I know it's just a one off route, but we should define it like the others.

tortillaj commented on lib/Server.js:
```diff
@@ -77,6 +75,9 @@ Server.prototype.setup = function() {
   app.get('/asset/:bucket/:assetName', auth, this.routes.serveFileAsset);
+  app.get('/asset-size/:bucket/:assetName', auth, this.routes.listAssetSize);
+  app.get('/buckets/:bucket/assets', auth, this.routes.assetMetadataByBucket);
+  app.del('/buckets/:bucket/asset/:assetName', auth, this.routes.deleteAssetFromBucket);
```
`app.del` is deprecated in favor of `app.delete`

tortillaj commented:
I'd like to provide a suggestion on the migrations too. It might be nice to include some kind of feedback to the console to inform if things are working ... and maybe count the files in the old file directory to compare against the new, reciphered file directory

lliss commented:
> I'd like to provide a suggestion on the migrations too. It might be nice to include some kind of feedback to the console to inform if things are working ... and maybe count the files in the old file directory to compare against the new, reciphered file directory

That's a good idea.

lliss commented:
I added another script that cleans up bad keys if they still exist after the other migrations. I also addressed all the feedback noted above. Can we give it another pass and then merge?

tortillaj commented:
@tizzo this looks good to me!

tortillaj commented:
oops - side note, I did not test AWS uploads!
